src/api/CommentApi.php

```diff
@@ -186,68 +194,72 @@ public function getActions(): array {
 	 * Adds a comment to a user's Curse Profile page or adds a new section on their talk page,
 	 * depending on what the user has chosen as their default user page.
 	 */
-	public function doAddToDefault() {
-		$user = MediaWikiServices::getInstance()->getUserFactory()->newFromId( $this->getMain()->getVal( 'user_id' ) );
-		if ( !$user || $user->isAnon() ) {
-			return $this->dieWithError( [ 'comment-invaliduser' ] );
```

Comment:
Users created via `UserFactory::newFromId` will never return `true` on `isAnon` calls, because `userId` is set (when creating user object).
`UserIdentityLookup:: getUserIdentityByUserId` performs actual check in cache/actor table